### PR TITLE
[Consensus] Add counters for number of ordered blocks in the pipeline

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

This will help understand the bottleneck in our pipeline

### Test Plan

Ran Forge and made sure the metrics make sense.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4241)

### CODE REVIEW

File: consensus/src/experimental/buffer_manager.rs [hunk 1]
```diff
@@ -190,6 +191,7 @@ impl BufferManager {
             self.buffer.len() + 1,
         );
         counters::NUM_BLOCKS_IN_PIPELINE.add(ordered_blocks.len() as i64);
+        counters::NUM_ORDERED_BLOCKS_IN_PIPELINE.add(ordered_blocks.len() as i64);
```
Comment (zekun000) [inline consensus/src/experimental/buffer_manager.rs:194]: hmm, this feels redundant to the one above, if we want to check whether it's executed or ordered, we should just add `status` to the same metrics

Comment (sitalkedia) [reply consensus/src/experimental/buffer_manager.rs:194]: Hmm..adding status on the metrics will be confusing as we want to track both number of blocks in the pipeline and number of ordered blocks in the pipeline. They are not mutually exclusive - so adding a status dimension won't work? Am I missing something?

**Review by igor-aptos — APPROVED** [review]
I am fine with this.

I think alternative is to split the pipeline into exclusively different statuses, and not have NUM_BLOCKS_IN_PIPELINE for the total

**Comment by sitalkedia** [comment]
Closing this in favor of https://github.com/aptos-labs/aptos-core/pull/4357

**Pull request closed without merging**